src/org/parosproxy/paros/model/SiteMap.java

```diff
@@ -745,5 +746,49 @@ private void clearParentFilter (SiteNode parent) {
 			clearParentFilter(parent.getParent());
 		}
 	}
+	
+	public void removeNodeFromParent(SiteNode node) {

```

Comment:
Will do. Removing the override and switching back to SiteNode for the param was just a change that I guess I lost track off while dealing with the null parent issue. Which as you pointed out was a time of use issue (can't get the parent from the node if the node has already been removed oops).